File: basic_roscpp/src/simple_node_pub.cpp
// Copyright 2020 Juan Carlos Manzanares Serrano

#include "ros/ros.h"
#include "std_msgs/Int64.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_node_pub");
    ros::NodeHandle n;

    ros::Publisher pub = n.advertise<std_msgs::Int64>("/contador", 1);

    ros::Rate loop_rate(10);

    int count = 0;

    while (ros::ok())
    {
        std_msgs::Int64 msg;
        msg.data = count++;

        pub.publish(msg);

        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: basic_rospy/src/simple_node_pub.py
#! /usr/bin/env python3

import rospy
from std_msgs.msg import Int32

def main():
    rospy.init_node("simple_node_pub")

    pub = rospy.Publisher("/contador", Int32, queue_size=1)

    msg = Int32()
    msg.data = 0

    rate = rospy.Rate(2)
    while(not rospy.is_shutdown()):
        pub.publish(msg)
        msg.data += 1
        rate.sleep()

    rospy.spin()

if __name__ == "__main__":

    try:
        main()
    except rospy.ROSInterruptException:
        pass
